File: 1556-make-two-arrays-equal-by-reversing-subarrays/make-two-arrays-equal-by-reversing-subarrays.c
int* sort(int *nums,int n)
{
    int t;
    for(int i=0;i<n;i++)
    {
        for(int j=0;j<n-i-1;j++)
        {
            if(nums[j]>nums[j+1])
            {
                t=nums[j];
                nums[j]=nums[j+1];
                nums[j+1]=t;
            }
        }
    }
    return nums;
}
bool canBeEqual(int* target, int targetSize, int* arr, int arrSize) {
    int *t,*a,c=0,ts=targetSize,as=arrSize;
    t=(int *)malloc(targetSize*sizeof(int));
    a=(int *)malloc(arrSize*sizeof(int));
    t=sort(target,ts);
    a=sort(arr,as);
    for(int i=0;i<targetSize;i++)
    {
        if(t[i]==a[i])
        c++;
    }
    if(c==arrSize)
    return true;
    else
    return false;
}

File: 1556-make-two-arrays-equal-by-reversing-subarrays/make-two-arrays-equal-by-reversing-subarrays.java
class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
        // Create an array to count the frequency of each number
        int[] count = new int[1001];  // Since 1 <= target[i], arr[i] <= 1000
        
        // Count the frequency of numbers in target (add) and arr (subtract)
        for (int i = 0; i < target.length; i++) {
            count[target[i]]++;
            count[arr[i]]--;
        }
        
        // Check if all frequencies are zero
        for (int c : count) {
            if (c != 0) {
                return false;
            }
        }
        
        return true;
    }
}